File: common/src/main/java/dev/number6/slack/dagger/HttpModule.java
package dev.number6.slack.dagger;

import dev.number6.slack.adaptor.OkHttpAdaptor;
import dev.number6.slack.port.HttpPort;
import dev.number6.slack.port.SecretsPort;
import dagger.Module;
import dagger.Provides;
import okhttp3.Call;
import okhttp3.OkHttpClient;

@Module(includes = {AWSSecretsModule.class})
public class HttpModule {

    @Provides
    public HttpPort providesHttpPort(Call.Factory client, SecretsPort secrets) {
        return new OkHttpAdaptor(client, secrets);
    }

    @Provides
    public Call.Factory callFactory() {
        return new OkHttpClient();
    }
}

File: common/src/main/kotlin/dev/number6/slack/dagger/HttpModule.kt
package dev.number6.slack.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.adaptor.OkHttpAdaptor
import dev.number6.slack.port.HttpPort
import dev.number6.slack.port.SecretsPort
import okhttp3.Call
import okhttp3.OkHttpClient

@Module(includes = [AWSSecretsModule::class])
class HttpModule {
    @Provides
    fun providesHttpPort(client: Call.Factory, secrets: SecretsPort): HttpPort {
        return OkHttpAdaptor(client, secrets)
    }

    @Provides
    fun callFactory(): Call.Factory {
        return OkHttpClient()
    }
}